### PR TITLE
[1.12] Remove some deprecated methods

### PR DESCRIPTION
**Pull request by bs2609**
Now we have a major version change, it's a chance to clean up some old deprecated methods that shouldn't be used.

This is initially just methods from my past PRs. Other contributors, are there any methods that can be safely added to this?

### CODE REVIEW

**Comment by liach** [comment]
You should search for todos and `@Deprecared` annotations.

**Comment by bs2609** [comment]
Paging the following for various PRs that deprecate things:

@mezz  #3839, #3689 

@williewillus  #3789, #3744, 

@bogger33 #3747 

@Choonster #3736 

@jcranky #3717 

@diesieben07 #3645 

@maxanier #3579 

I will go through and add some of these myself, this serves as notification if anyone wants to comment, and as a helpful reference for anyone reviewing this.

**Comment by bs2609** [comment]
Also going to remove `Java6Utils`, as it's no longer needed.

**Comment by Choonster** [comment]
The deprecated `BonemealEvent` methods and constructors should be fine to remove.

**Comment by bogger33** [comment]
Same with `getFluidHeightForRender(IBlockAccess world, BlockPos pos)`

**Comment by diesieben07** [comment]
`AttachCapabilityEvent.Item` can go. That means that `AttachCapabilityEvent.caps` can be private again.

**Comment by bs2609** [comment]
Ok, `AttachCapabilityEvent.Item` has been removed, @diesieben07, does 5cbf5b5afe62c59342fc843676839d0ba8aa9b58 look alright to you?

**Comment by maxanier** [comment]
Both things that were deprecated by my PR (#3579) are already removed in 1.12 as far as I can tell.

**Comment by Choonster** [comment]
`ItemDye.applyBonemeal(ItemStack, World, BlockPos, EntityPlayer)` can also be removed.

**Comment by SteveKunG** [comment]
Same with this? #3512

**Comment by bs2609** [comment]
Should the various -action suffixed methods in ForgeHooks just inherit the original names now?

**Comment by williewillus** [comment]
@bs2609 yes, those were there to add #3744 without breaking compat (if I'm thinking of the same methods you are)

File: patches/minecraft/net/minecraft/item/ItemDye.java.patch [hunk 1]
```diff
@@ -27,15 +27,6 @@
...
-+    /**
-+     * @deprecated Use {@link #applyBonemeal(ItemStack, World, BlockPos, EntityPlayer, EnumHand)} instead.
-+     */
-+    @Deprecated
-+    public static boolean applyBonemeal(ItemStack p_179234_0_, World p_179234_1_, BlockPos p_179234_2_, EntityPlayer player)
-+    {
-+        return applyBonemeal(p_179234_0_, p_179234_1_, p_179234_2_, player, null);
-+    }
-+
 +    public static boolean applyBonemeal(ItemStack p_179234_0_, World p_179234_1_, BlockPos p_179234_2_, EntityPlayer player, @javax.annotation.Nullable EnumHand hand)
```
Comment (liach) [inline patches/minecraft/net/minecraft/item/ItemDye.java.patch:30]: You probably should remove the nullable on the hand.

Comment (Choonster) [reply patches/minecraft/net/minecraft/item/ItemDye.java.patch:30]: Why? The hand is supposed to be nullable, we went over this in #3736.

Comment (liach) [reply patches/minecraft/net/minecraft/item/ItemDye.java.patch:30]: ok then

**Review by mezz — APPROVED** [review]

**Comment by bs2609** [comment]
Removed the deprecated mod GUI methods. This resolves #3512.

@cpw, you deprecated some of this code in 543156a008295efb43f155c40e8aeb5db6080e6c - is this removal ok?

**Comment by mezz** [comment]
I think you should go ahead and remove `RuntimeOptionGuiHandler` but leave the world compatibility code for now.

File: src/main/java/net/minecraftforge/common/ForgeHooks.java [hunk 2]
```diff
@@ -299,164 +299,12 @@ public ItemStack getStack(Random rand, int fortune)
 
     /**
      * Called when a player uses 'pick block', calls new Entity and Block hooks.
+     *
+     * Note: replaces most of {@link Minecraft#middleClickMouse()},
+     * refer to the clean version of that method for implementation requirements.
      */
     public static boolean onPickBlock(RayTraceResult target, EntityPlayer player, World world)
     {
-        /*
```
Comment (mezz) [inline src/main/java/net/minecraftforge/common/ForgeHooks.java:305]: Please keep this commented out section, it's needed for updating.

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by mezz** [comment]
I'm going to manually resolve and merge this one now so it makes it in before the breaking-changes deadline.

**Comment by mezz** [comment]
Merged here https://github.com/MinecraftForge/MinecraftForge/commit/9a09ff68aac457d4297355bd9cb30286af92ec05

**Pull request closed without merging**